neo/Network/P2P/Payloads/Transaction.cs

```diff
@@ -92,8 +94,8 @@ public void DeserializeUnsigned(BinaryReader reader)
             if (SystemFee + NetworkFee < SystemFee) throw new FormatException();
             ValidUntilBlock = reader.ReadUInt32();
             Attributes = reader.ReadSerializableArray<TransactionAttribute>(MaxTransactionAttributes);
-            var cosigners = Attributes.Where(p => p.Usage == TransactionAttributeUsage.Cosigner).Select(p => new UInt160(p.Data)).ToArray();
-            if (cosigners.Distinct().Count() != cosigners.Length) throw new FormatException();
+            Cosigners = reader.ReadSerializableArray<Cosigner>();
+            if (Cosigners.Select(u => u.Account).Distinct().Count() != Cosigners.Length) throw new FormatException();
```

Comment:
@erikzhang @shargon does this allow multiple repeated Account (but "distinct" cosigners? or is it also intended to only allow unique "account" cosigners here (with no combination)?)

Reply:
This need to be tested, yes

Reply:
Ok, tested. It indeed fails if repeated account is put there, even if scopes are different. It's reasonable to me.

Reply:
So I can sign for two contracts in different group, buy i don't sign for all of them

``` [Sign] ContractA-> ContractB -> [Sign] ContractC ```